migrate/loader.go

```diff
@@ -0,0 +1,137 @@
+package migrate
+
+import (
+	"fmt"
+	"github.com/packwiz/packwiz/core"
+	"github.com/spf13/cobra"
+	"golang.org/x/exp/slices"
+	"os"
+	"strings"
+)
+
+var loaderCommand = &cobra.Command{
+	Use:   "loader",
+	Short: "Migrate your loader versions to newer versions.",
+	Args:  cobra.ExactArgs(1),
+	Run: func(cmd *cobra.Command, args []string) {
+		modpack, err := core.LoadPack()
+		if err != nil {
+			// Check if it's a no such file or directory error
+			if os.IsNotExist(err) {
+				fmt.Println("No pack.toml file found, run 'packwiz init' to create one!")
+				os.Exit(1)
+			}
+			fmt.Printf("Error loading pack: %s\n", err)
+			os.Exit(1)
+		}
+		// Get our current loader
+		currentLoader := modpack.GetLoaders()
+		// Do some sanity checks on the current loader slice
+		if len(currentLoader) == 0 {
+			fmt.Println("No loader is currently set in your pack.toml!")
+			os.Exit(1)
+		} else if !slices.Contains(currentLoader, "quilt") && len(currentLoader) > 1 {
+			fmt.Println("You have multiple loaders set in your pack.toml, this is not supported!")
+			os.Exit(1)
+		} else if slices.Contains(currentLoader, "quilt") {
+			// We have quilt, so we need to remove fabric from the loaders list
+			fabricIndex := slices.Index(currentLoader, "fabric")
+			currentLoader = slices.Delete(currentLoader, fabricIndex, fabricIndex+1)
+		}
+		// Get the Minecraft version for the pack
+		mcVersion, err := modpack.GetMCVersion()
+		if err != nil {
+			fmt.Printf("Error getting Minecraft version: %s\n", err)
+			os.Exit(1)
+		}
+		if args[0] == "latest" {
+			fmt.Println("Updating to latest loader version")
+			// We'll be updating to the latest loader version
+			for _, loader := range currentLoader {
+				_, latest, gottenLoader := getVersionsForLoader(loader, mcVersion)
+				if !updatePackToVersion(latest, modpack, gottenLoader) {
+					continue
+				}
+				// Write the pack to disk
+				err = modpack.Write()
+				if err != nil {
+					fmt.Printf("Error writing pack.toml: %s\n", err)
+					continue
+				}
+			}
+		} else if args[0] == "recommended" {
+			// TODO: Figure out a way to get the recommended version, this is Forge only
+			fmt.Println("Currently updating to the recommended loader version is not supported!")
+			os.Exit(1)
+		} else {
+			fmt.Println("Updating to explicit loader version")
+			// This one is easy :D
+			versions, _, loader := getVersionsForLoader(currentLoader[0], mcVersion)
+			// Check if the loader happens to be Forge, since there's two version formats
+			if loader.Name == "forge" {
+				var wantedVersion string
+				// Check if we have a "-" in the version
```

Comment:
I think the version list getter already strips versions for you? The init command doesn't seem to do any post-processing anyway.

Reply:
This is input from the user, which'll let them either do `1.19.4-44.1.23` or `44.1.23` for example, since Forge uses both formats interchangeable everywhere.

Reply:
Fair... this could probably be extracted into a `cmdshared` function and used in the init command too

Reply:
Abstracted out into `cmdshared.GetRawForgeVersion`.